physical/s3/s3.go

```diff
@@ -188,16 +188,20 @@ func (s *S3Backend) Get(ctx context.Context, key string) (*physical.Entry, error
 	if resp == nil {
 		return nil, fmt.Errorf("got nil response from S3 but no error")
 	}
+	defer resp.Body.Close()
```

Comment:
Might it be more defensive to do this right before line 178? Something like, `if resp != nil { defer resp.Body.Close() }`? Then there'd be absolutely no way there could be a memory leak if we hit 404's a lot, and if for some reason the AWS client still returned a body. I know for the `http.Client` that doesn't happen, but I don't know if the AWS client maintains the same contract. I can't find anything saying either way in their docs.

Reply:
That's fair, considering we're thinking of returning data with 404s. I'll update.